test/test-network/systemd-networkd-tests.py

```diff
@@ -6839,6 +6839,48 @@ def check(self, ipv4, ipv6):
         check(self, True, False)
         check(self, False, True)
         check(self, False, False)
+    
+    def test_dhcp_client_default_use_domains(self):        
+        def check(self, ipv4, ipv6):
+            os.makedirs(networkd_conf_dropin_dir, exist_ok=True)
+            with open(os.path.join(networkd_conf_dropin_dir, 'default_use_domains.conf'), mode='w', encoding='utf-8') as f:
+                f.write('[DHCPv4]\nUseDomains=')
+                f.write('yes' if ipv4 else 'no')
+                f.write('\n[DHCPv6]\nUseDomains=')
+                f.write('yes' if ipv6 else 'no')
+            
+            check_output('systemctl daemon-reload')
+            restart_networkd()
+            check_output('systemctl restart systemd-resolved.service')
+
+            self.wait_online('veth99:routable')
+
+            # link becomes 'routable' when at least one protocol provide an valid address. Hence, we need to explicitly wait for both addresses.
+            self.wait_address('veth99', r'inet 192.168.5.[0-9]*/24 metric 1024 brd 192.168.5.255 scope global dynamic', ipv='-4')
+            self.wait_address('veth99', r'inet6 2600::[0-9a-f]*/128 scope global (dynamic noprefixroute|noprefixroute dynamic)', ipv='-6')
+
+            output = resolvectl('domain', 'veth99')
```

Comment:
This may be racy, as even if networkd acquires DHCPv{4,6} leases, resolved may not be read DNS related settings yet. Hence, below check needs to be run several times something like the following:
```python
for _ in range(20):
        output = resolvectl('domain', 'veth99')
        if ipv4 or ipv6:
                if 'example.com' in output:
                        break
        else:
                if 'example.com' not in output:
                        break
        time.sleep(0.5)
else:
        print(output)
        self.fail('unexpected domain setting in resolved...')
```